### ISSUE TITLE
Depending on old sodium-native means building from source in node 12+

### ISSUE THREAD

**Issue by jon-v** [emilbayes/secure-password#20]
Would be nice if secure-password could be updated to depend on the latest sodium-native (2.4.6).

**Comment by emilbayes** [comment]
That is a minor version upgrade and the package.json here allows for that semver range. Are you sure that's what's causing recompiles?

**Comment by nexdrew** [comment]
I'm currently migrating an app with a package-lock from Node 10 to Node 12, and I was running into recompile issues as well when running `npm ci` (which only reads from package-lock).

However, the problems disappeared once I was able to bump the transitive sodium-native dependency to its latest version (from 2.2.1 to 2.4.6 at time of this writing) in my package-lock.

I did this by first uninstalling secure-password (`npm uninstall secure-password`) and then re-installing it (`npm i -P secure-password`), which allowed the npm CLI to re-reconcile the `^2.2.1` semver range to 2.4.6 and bake that into my package-lock. I don't know if there's a better way to tell the npm CLI to manually upgrade a transitive dependency in package-lock, but I think this is a reliable way to do it.

Just commenting for the sake of others, hopefully this helps.
